content/open-source/sigstore/cosign/how-to-verify-file-signatures-with-cosign.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "How to Verify File Signatures with Cosign"
+type: "article"
+description: "Use Cosign to verify non-container software artifacts"
+lead: "Cosign can verify software artifacts beyond containers"
+date: 2022-12-21T15:22:20+01:00
+lastmod: 2022-12-21T15:22:20+01:00
+draft: false
+images: []
+menu:
+  docs:
+    parent: "cosign"
+weight: 005
+toc: true
+---
+
+Cosign can verify more than just containers. Blobs, or binary large objects, and standard files can be verified in a similar way. You can publish a blob or other artifact to an OCI (Open Container Initiative) or elsewhere and then use Cosign to verify its signature. This tutorial assumes you have Cosign installed, which you can achieve by following our [How to Install Cosign guide](../how-to-install-cosign/).
+
+We’ll verify a binary artifact, in this case, a copy of [`apko`](/open-source/apko/overview/), which is a command-line tool for building container images using a declarative language based on YAML. The methods in this tutorial apply to any blob file Cosign has signed with a keyless signature.
+
+### Verifying a binary with Cosign keyless signatures
+
+All `apko` releases are released with [keyless signatures using Cosign](https://edu.chainguard.dev/open-source/sigstore/cosign/an-introduction-to-cosign/#keyless-signing). You can verify the signature for an `apko` release using the `cosign` tool directly, or by calculating the SHA256 hash of the release and finding the corresponding Rekor transparency log entry.
+
+If you would like to learn how to verify a binary using Rekor or curl, follow the steps in our guide [How to Verify File Signatures with Rekor or curl](open-source/sigstore/rekor/how-to-verify-file-signatures-with-rekor-or-curl/).
+
+We'll use the `apko_0.6.0_linux_arm64.tar.gz` tar archive from the apko [GitHub Release v0.6.0 page](https://github.com/chainguard-dev/apko/releases/tag/v0.6.0) in this example.
```

Comment:
Maybe we can say something about the general release page to verify other versions

Reply:
Good idea, will add a note that this will work for any release with a signature and certificate

Reply:
We can come back to this